explorer/interpreter/interpreter.cpp

```diff
@@ -860,6 +865,42 @@ auto Interpreter::Convert(Nonnull<const Value*> value,
       }
       return Convert(value, destination_type, source_loc);
     }
+    case Value::Kind::PointerValue: {
+      if (destination_type->kind() != Value::Kind::PointerType ||
+          cast<PointerType>(destination_type)->type().kind() !=
+              Value::Kind::NominalClassType) {
+        // No conversion needed
+        return value;
+      }
+
+      // Get pointed value
+      const auto* src_ptr = cast<PointerValue>(value);
+      CARBON_ASSIGN_OR_RETURN(const auto* pointee,
+                              heap_.Read(src_ptr->address(), source_loc))
+      if (pointee->kind() == Value::Kind::NominalClassType) {
+        // TODO: When does that happen? Just return as-is.
```

Comment:
Is this a question because you're not sure and think it won't occur? If so, you might instead CARBON_CHECK -- i.e., if this code shouldn't run, add a CHECK and if someone figures out code that causes it, then it's done.

Reply:
The question is a real question actually :D, I was hoping you or @zygoloid might know. I saw it happen, and don't understand what this represents (maybe during type-checking only?). Killing this branch leads to errors. Either way, something to figure out before merging.

Reply:
Without seeing the errors, I would expect this branch needs to be adjusted. What's the destination_type?

This might be a type checking thing, i.e. `A*` being PointerType -> NominalClassType. But It should probably still depend on what the destination_type is: destination_type there is probably either "Type" or "PointerType -> Type".

Reply:
Tested the branch again, and it isn't used anymore, I think this was caused by `BaseElement` previously returning a different type. Fixed, thanks for suggestions.